src/app/pages/ArticlePage/ArticlePage.tsx

```diff
@@ -131,6 +139,22 @@ const ArticlePage = ({ pageData }: { pageData: Article }) => {
     ...(isCPS && { pageTitle: `${atiAnalytics.pageTitle} - ${brandName}` }),
   };
 
+  let blocksWithExperimentTopStories = blocks;
+  const topStoriesContent = pageData?.secondaryColumn?.topStories;
+  const shouldEnableExperimentTopStories =
+    enableExperimentTopStories({
+      isAmp,
+      service,
+      pathname,
+    }) && topStoriesContent;
+
+  if (shouldEnableExperimentTopStories) {
+    blocksWithExperimentTopStories = insertExperimentTopStories({
+      blocks,
+      topStoriesContent,
+    });
+  }
```

Comment:
**Suggestion (non-blocking)** Move this into the experiment top stories helper you've already created and just return either the blocks unchanged or the blocks with the top stories moved (as well as other things you might need)? This would hide the implementation detail of this experiment from anyone who is just parsing the article page code.